packages/phone-number-privacy/signer/src/signer/domain/quota.service.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  checkSequentialDelayRateLimit,
+  DomainQuotaStatusRequest,
+  DomainRestrictedSignatureRequest,
+  ErrorMessage,
+  isSequentialDelayDomain,
+} from '@celo/phone-number-privacy-common'
+import { Transaction } from 'knex'
+import { DomainStateRecord } from '../../database/models/domainState'
+import {
+  getDomainStateRecordOrEmpty,
+  updateDomainStateRecord,
+} from '../../database/wrappers/domainState'
+import { IQuotaService, OdisQuotaStatusResult } from '../quota.interface'
+import { DomainSession } from './session'
+
+declare type QuotaDependentDomainRequest =
+  | DomainQuotaStatusRequest
+  | DomainRestrictedSignatureRequest
+
+export class DomainQuotaService implements IQuotaService<QuotaDependentDomainRequest> {
+  public async checkAndUpdateQuotaStatus(
+    state: DomainStateRecord,
+    session: DomainSession<QuotaDependentDomainRequest>,
+    trx: Transaction<DomainStateRecord>,
+    attemptTime: number = Date.now() / 1000 // Convert current time in ms to seconds.
+  ): Promise<OdisQuotaStatusResult<QuotaDependentDomainRequest>> {
```

Comment:
Similar comment to elsewhere
```suggestion
    attemptTime?: number
  ): Promise<OdisQuotaStatusResult<QuotaDependentDomainRequest>> {
    attemptTime = attemptTime ?? Date.now() / 1000 // Convert current time in ms to seconds.
```